arches/app/models/tile.py

```diff
@@ -557,8 +565,16 @@ def flatten_tiles(obj):
         return tiles
 
     def after_update_all(self):
-        nodegroup = models.NodeGroup.objects.get(pk=self.nodegroup_id)
-        for node in nodegroup.node_set.all():
+        try:
+            nodegroup = SimpleNamespace(**next((x for x in self.serialized_graph["nodegroups"] if x["nodegroupid"] == self.nodegroup_id), None))
+                    
+            nodes = [(node for node in self.serialized_graph["nodes"] if node["nodegroup_id"] == self.nodegroup_id)]
+            nodes = [SimpleNamespace(**next(node, None)) for node in nodes]
+        except:
+            nodegroup = models.NodeGroup.objects.get(pk=self.nodegroup_id)
+            nodes = nodegroup.node_set.all()
```

Comment:
This one should stay; this is meant to catch if serialized graph is None - in which case the nodegroup wouldn't be set.  But the catch should be more explicit so that the typeerror is caught.

Reply:
Yep I'm with you on why the except clause is here, I'm just suggesting rewriting the content of the except clause.

> in which case the nodegroup wouldn't be set

I didn't follow this part -- Django will lazily fetch `nodegroup`, so it's better to rely on that than always doing the extra query AFAIU.